### PR TITLE
Quiz Creation Select Resources - Keyword search

### LINKED ISSUE #11786: Quiz Creation Select Resources - Implement Keyword search.
### Overview 
We need a search functionality on the Resource selection to allow users to search resources  by keyword in the topic tree  as included in[ the design](https://www.figma.com/file/1tQQ0Sw1yDjzO66hyjLVPM/Kolibri-UX-Fixes?type=design&node-id=2159-79951&mode=design&t=OMO9ukhccZsnFa4D-0). If in doubt, reach out to @tomiwaoLE for clarity.  
### Guidance 
 What we need to add is that the search  should be integrated   into the router similarly to how we've differentiated between Bookmarks & normal topic navigation (basically, we'll want a /search route in addition to what we've added)
The search was handled previously in Vuex, largely by way of this [handler](https://github.com/learningequality/kolibri/blob/6b8acac91c4d2d476d7c6f7b5f87a1a42d6ba521/kolibri/plugins/coach/assets/src/modules/examCreation/handlers.js#L274) -- we will not be using handlers anymore, so the same API call & logic there should be migrated to a new module  `useFetchTree `which gives an example of how to handle the "fetching more" 

Using LessonSearchBox we can leverage the[ this.$route.params.searchTerm](https://github.com/learningequality/kolibri/blob/66589c69dd0c59c443acf33e54a062ff89af99d7/kolibri/plugins/coach/assets/src/views/plan/LessonResourceSelectionPage/index.vue#L234) in the [setup function ](https://github.com/learningequality/kolibri/blob/66589c69dd0c59c443acf33e54a062ff89af99d7/kolibri/plugins/coach/assets/src/views/plan/CreateExamPage/ResourceSelection.vue#L117)in ResourceSelection. The the only pain is going to be migrating   the API calls from the old Vuex handler,  and to make sure that contentList is set to show the results.

 #### Acceptance criteria
 

- [ ] 1. Migrate the old from Vuex handler to the useFetchTree module to handle data fetching. 
- [ ] 2. Add a  /search route to the router, similar to the bookmarks route.  You can confirm this is working by testing against the API rather than within a component. Even using something like console logging to see the object is returned would work.)
- [ ] 3a. Utilize the existing LessonSearchBox component within ResourceSelection and leverage the this.$route.params.searchTerm in the setup function to access search terms.  (Now we are implementing actually into the frontend)
 - [ ] 3b. When a user exit the search and should route back to ResourceSelctionPage

### PR DESCRIPTION
**Pull request by AlexVelezLl**
## Summary

This PR adds a search input in Quiz Creation Select resources Side Panel.

**Preview**

https://github.com/learningequality/kolibri/assets/51239030/b8edd60a-df8c-4821-a00e-a2f2935ffb20

## References

closes #11786

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [x] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [ ] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/plugins/coach/assets/src/views/plan/CreateExamPage/SectionSidePanel.vue
Comment (AlexVelezLl) [inline kolibri/plugins/coach/assets/src/views/plan/CreateExamPage/SectionSidePanel.vue:109]: For some reason there was a case where `workingResourcePool` was undefined here and it didn't let me move forward

**Review by nucleogenesis — COMMENTED** [review]
This looks great @AlexVelezLl -  I'm going to list a few thoughts here but won't approve or request changes quite yet.

- The changes I've made to the routing in #11862 fix most (if not all) of the bugs I'm seeing when testing this (ie, cannot save changes after adding search result, back button weirdness, 
- One thing that we didn't consider while working on this was setting the local loading state whenever we're fetching search results. When we run this locally it's so fast that it's hard to see how bad the UX would be when the load takes a few seconds. _That said, however_, I think that this is an issue we can better handle in a follow-up as this is not solely a problem w/ the search loading but with all of the loading in the side panel right now.

Great work - I'm going to wait for review on my latest changes to the PR I noted above and if that goes through, I think I'll merge this PR and then fix conflicts any conflicts it creates in my PR before merging that.

**update** - I've added a link to the project tech debt tracker. Updating how the loading states are managed will jive well with a general follow-up in how we manage the loading states altogether which I will work on after wrapping up question replacement early next week.

File: kolibri/plugins/coach/assets/src/views/plan/CreateExamPage/ResourceSelection.vue
Comment (rtibbles) [inline kolibri/plugins/coach/assets/src/views/plan/CreateExamPage/ResourceSelection.vue]: Unless the component is being torn down and recreated when the searchQuery updates (which I think we are trying to avoid happen), this value will not recompute.

You may want to instead make the fetchMoreQuizResources function behave conditionally on the searchQuery instead.

Also, it should be `searchQuery.value` I believe that we conditionalize on?

**Review by rtibbles — DISMISSED** [review]
One thing that looks suspect to me - conditionalizing setup exports will not let the component be reactive to changes in searchQuery state.

**Comment by AlexVelezLl** [comment]
Thanks @rtibbles! I have updated it :open_hands:

**Review by nucleogenesis — APPROVED** [review]
Thanks for the updates and comments Alex & Richard.

**Pull request merged**
